src/mlpack/bindings/go/get_type.hpp

```diff
@@ -0,0 +1,132 @@
+/**
+ * @file get_type.hpp
+ * @author Yasmine Dumouchel
+ *
+ * Template metaprogramming to return the string representation of the type
+ * for the Go bindings.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_BINDINGS_GO_GET_TYPE_HPP
+#define MLPACK_BINDINGS_GO_GET_TYPE_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/util/is_std_vector.hpp>
+
+namespace mlpack {
+namespace bindings {
+namespace go {
+
+template<typename T>
+inline std::string GetType(
+    const util::ParamData& /* d */,
+    const typename boost::disable_if<util::IsStdVector<T>>::type* = 0,
+    const typename boost::disable_if<data::HasSerialize<T>>::type* = 0,
+    const typename boost::disable_if<arma::is_arma_type<T>>::type* = 0)
+{
+  return "unknown";
+}
+
+template<>
+inline std::string GetType<int>(
+    const util::ParamData& /* d */,
+    const typename boost::disable_if<util::IsStdVector<int>>::type*,
+    const typename boost::disable_if<data::HasSerialize<int>>::type*,
+    const typename boost::disable_if<arma::is_arma_type<int>>::type*)
+{
+  return "Int";
```

Comment:
It is use to write functions of .go file.

Reply:
Ohh!  Now I see.  It's used in calls like

```
    std::cout << prefix << prefix << "gonumToArma" << GetType<T>(d)
              << "(\"" << d.name << "\", param." << goParamName
              << ")" << std::endl;
```

so all of these types are correct.  Now it makes sense.  Sorry about that confusion (I'll resolve all the other related comments).  But do you think that we could update the comment for `GetType`?  Currently it reads:

```
/**
 * Print an option into a std::string.  This should print a short, one-line
 * representation of the object.  The string will be stored in the output
 * pointer.
 *
 * @param data Parameter data struct.
 * @param input Unused parameter.
 * @param output Output storage for the string.
 */
```

Maybe it would be better to say something like that the function is used to generate the type names that are used in calls to functions like `gonumToArma<type>()` or `setParam<type>()` or similar, and therefore what's returned isn't *exactly* the Go native type used for that parameter type.

Reply:
Updated the comment.